src/screens/Swap/FormSelection/SelectAccountScreen.js

```diff
@@ -0,0 +1,253 @@
+/* @flow */
+
+import React, { useCallback, useMemo } from "react";
+import {
+  View,
+  StyleSheet,
+  FlatList,
+  SafeAreaView,
+  TouchableOpacity,
+} from "react-native";
+import { useSelector } from "react-redux";
+import { Trans } from "react-i18next";
+import { useTheme } from "@react-navigation/native";
+
+import {
+  accountWithMandatoryTokens,
+  flattenAccounts,
+} from "@ledgerhq/live-common/lib/account/helpers";
+import { getAccountCurrency } from "@ledgerhq/live-common/lib/account";
+
+import type { SearchResult } from "../../../helpers/formatAccountSearchResults";
+import { accountsSelector } from "../../../reducers/accounts";
+import { TrackScreen } from "../../../analytics";
+import LText from "../../../components/LText";
+import FilteredSearchBar from "../../../components/FilteredSearchBar";
+import AccountCard from "../../../components/AccountCard";
+import KeyboardView from "../../../components/KeyboardView";
+import { formatSearchResults } from "../../../helpers/formatAccountSearchResults";
+import { NavigatorName, ScreenName } from "../../../const";
+
+import type { SwapRouteParams } from "..";
+import AddIcon from "../../../icons/Plus";
+import { swapSelectableCurrenciesSelector } from "../../../reducers/settings";
+
+const SEARCH_KEYS = ["name", "unit.code", "token.name", "token.ticker"];
+
+type Props = {
+  navigation: any,
+  route: { params: SwapRouteParams },
+};
+
+export default function SelectAccount({ navigation, route }: Props) {
+  const { colors } = useTheme();
+  const { swap, target, selectedCurrency, setAccount, provider } = route.params;
+
+  const unfilteredAccounts = useSelector(accountsSelector);
+  const selectableCurrencies = useSelector(swapSelectableCurrenciesSelector);
+
+  const accounts = unfilteredAccounts.filter(
+    acc => acc && selectableCurrencies.includes(getAccountCurrency(acc).id),
+  );
+
+  const enhancedAccounts = useMemo(() => {
```

Comment:
`useMemo` seems useless here (unless I'm mistaken) since the `accounts` dependency reference will change every time the component renders

Reply:
i'll memo the accounts as well then :grimacing: